### ISSUE TITLE
Split a wallet into two and/or remove accounts from a wallet.

### ISSUE THREAD

**Issue by destenson** [bitshares/bitshares-ui#264]
I have some wallets with "too many" accounts in them. Most of these accounts are not used anymore, so they don't need to be included in my normally active wallet. So, it would be very useful for wallet management if there were a way to split a wallet into two wallets, with the ability to select which wallet different accounts should go into. It should also be possible to remove accounts from a wallet.

Wallet management in general needs an overhaul badly. Some of the issues that I and others have brought up caused by having "too many" accounts, could be mitigated or addressed more easily if there were better wallet management capabilities.

**Comment by wmbutler** [comment]
So, you have say 3 wallets with 5 accounts each and you'd like to be able to arbitrarily move them from wallet to wallet correct, as well as delete accounts? Maybe you'd rather hide accounts. This issue is really broad in scope. It'll need to be broken into user stories that are bite sized. Maybe:

- Procedure to transfer account from one wallet to another
- Ability to hide accounts (we have this already but I do believe we are working an open bug on how well the hide feature works currently)

Beyond that, what else are you asking for?

**Comment by destenson** [comment]
More like, I have 1 wallet with 150 accounts, and I'd like to split it into two wallets, one with 25 accounts and the rest in the other. I would be open to doing this manually (and I have to some degree), but there doesn't seem to be any way of removing an account that I've already added to a different wallet.

I would be fine with hiding accounts, but that's not sufficient since, as you mentioned, it is broken (probably more than you're aware).

TBH, I'd like to ask that wallets with "too many" accounts be supported, but since it's not, I'm stuck with wallets (and versions of their backups) with too many accounts, or wallets with not too many but not the right ones, or wallets with only a few of the right ones but not all that I want.

Your suggestion to add a procedure for transferring an account from one wallet to another would be sufficient & greatly appreciated.

**Comment by wmbutler** [comment]
This is a transfer operation (moving an account from wallet a to wallet b). Then obviously we need a bug free way to hide accounts which is being worked. So I'd suggest you modify this issue something like: "Provide an easy way to transfer an account to a different wallet". We can then groom and refine this issue until it can go into a sprint.

**Comment by btsfav** [comment]
Personally wouldn't pay one cent from developers fund for this. account squatters should pay for this themselves, as it only affects 3-4 people.

**Comment by wmbutler** [comment]
I have reserved a number of accounts for friends and have managed on their behalf until they were ready to do so on their own. This is not just for squatters in my opinion. Users should be able to move accounts between wallets with minimal effort.

**Comment by destenson** [comment]
@btsfav, you make a good point. I hadn't thought of the feature assisting squatters. But, personally, I don't care who pays for it, or if it's a public feature, or what... I just want to be able to get many of these accounts out of my main wallet and into one I can monitor less often. This has been giving me frustration and a negative experience with bitshares for about a year, though I've only brought it up recently.

Honestly, I'd rather the underlying code be fixed so that it doesn't matter how many accounts a wallet has. As is, bitshares is completely unusable for the wallet I've been using for the last 14 months. I've had to manually copy my most important accounts into a new wallet for now, so it can at least use bitshares without the browser tab hanging. (BTW: I'm pretty sure some of the other bugs people have reported, even those without many accounts in their wallet, would be fixed by allowing any number of accounts or subscriptions.)

Instead of having it implemented in the main app, is there a manual procedure that someone could explain to me instead?

**Comment by landry314** [comment]
@destenson, are you copying the private keys manually then creating a new wallet and importing them there? This should create a new wallet with a small number of accounts.

Of course, then the old wallet will also have those accounts. Do you then switch to those accounts and remove the keys in Account --> Permissions? Does this remove those accounts from the old wallet?

I am wanting to do this myself as I only currently need about 3 accounts. Hiding is working ok but I am always hitting bugs and having to do a clean install. I am thinking this might help until all the wallet and account management bugs are squashed.

**Comment by destenson** [comment]
@landry314, yes, I've done that. That's my current workaround for the issue, but it prevents me from being able to monitor a number of accounts that I'd like to.

As for removing keys from the wallet. I haven't tried that in a long time, but I've not been successful removing any keys (except by hacking the browser's local DB where it stores them).. Once a wallet is too full and suffering from the problems associated with that, everything gets very flakey and untrustworthy and most of the time, simply doesn't work.

**Comment by clockworkgr** [comment]
I believe we need to sort this  one out soon. I've mentioned the need for a "clearer" management interface of the local wallet contents before.

This will need UX work. @gibbsfromncis / @nikandrovalex  Can you guys have a look to get an estimate?

My thoughts:

List of wallets stored in browser by name:
Under currently unlocked one, list (in table form) all accounts "accessible" through that wallet while mentioning the level of access (active/owner/memo) and whether access is explicit (account/private key pairs actually in wallet) or derived (via get_key or get_account_references).
Each row should have a  checkbox and a list of associated actions underneath the table.

E.g. I can check 5-6  accounts and click remove. or "export to new wallet", "move to other wallet" (dropdown)etc.

Similarly the locked wallet names should have a "switch to" button.

Password popups as required.

When exporting to new wallet you'd have to provide the existing wallet's pass to unlock the keys and a new pass for the new wallet.

I did a basic form of this for importing .bin files to Beet so here's a screenshot to get an initial idea. Obviously needs more data and actions.

![image](https://user-images.githubusercontent.com/6826762/52698479-34374b80-2f7c-11e9-820e-b13b0811cf8e.png)

We will also need an estimate for coding this.

Anyone else care to share more use cases/scenarios?
